### ISSUE TITLE
Framed not usable out of the box with streaming ServerProto variants 

### ISSUE THREAD

**Issue by aajtodd** [tokio-rs/tokio-proto#90]
The bound on [ServerProto::Transport](https://tokio-rs.github.io/tokio-proto/tokio_proto/streaming/pipeline/trait.ServerProto.html#associatedtype.Transport) means you can't just add a codec and use [Framed](https://tokio-rs.github.io/tokio-core/tokio_core/io/struct.Framed.html) out of the box with the streaming `ServerProto` variants.

`Framed` implements `Stream` and `Sink` but not `Transport` which comes from `tokio-proto`.

**Comment by aajtodd** [comment]
The respective `Transport` methods all have defaults so adding implementations of `streaming::pipeline::Transport` and `streaming::multiplex::Transport` for `Framed` would work but would mean users can't opt into handling those methods.

```rust
// pipeline
impl<T:Io + 'static, C: Codec + 'static> Transport for Framed<T,C> {}

// multiplex
impl<T:Io + 'static, C: Codec + 'static, B> Transport<B> for Framed<T,C> {}
```

**Comment by aturon** [comment]
I think these missing impls were just an oversight. There's not really any alternative way to implement the extra methods over `Framed`. I'd be happy to take a PR!

**Comment by carllerche** [comment]
Fixed by #93. Thanks
